commercial/app/model/commercial/Merchandise.scala

```diff
@@ -0,0 +1,370 @@
+    path.readNullable[Double] orElse path.readNullable[String].map(_.map(_.toDouble))
+  }
+
+  implicit val bookReads: Reads[Book] = (
+    (JsPath \ "name").read[String] and
+      authorReads and
+      (JsPath \ "isbn").read[String] and
+      stringOrDoubleAsDouble("regular_price_with_tax") and
+      stringOrDoubleAsDouble("final_price_with_tax") and
+      (JsPath \ "description").readNullable[String] and
+      (JsPath \ "images")(0).readNullable[String] and
+      (JsPath \ "product_url").readNullable[String] and
+      (JsPath \ "guardian_bestseller_rank").readNullable[String].map(_.map(_.toDouble.toInt)) and
+      ((JsPath \ "categories")(0) \ "name").readNullable[String] and
+      (JsPath \ "keywordIds").readNullable[Seq[String]].map(_ getOrElse Nil)
+    )(Book.apply _)
+
+  implicit val writesBook: Writes[Book] = Json.writes[Book]
+}
+
+object Masterclass {
+  private val guardianUrlLinkText = "Full course and returns information on the Masterclasses website"
+
+  def fromEvent(event: EBEvent): Option[Masterclass] = {
+
+    val doc: Document = Jsoup.parse(event.description)
+
+    def extractGuardianUrl: Option[String] = {
+      val elements :Array[Element] = doc.select(s"a[href^=http://www.theguardian.com/]:contains($guardianUrlLinkText)")
+          .toArray(Array.empty[Element])
+
+      elements.headOption match {
+        case Some(e) => Some(e.attr("href"))
+        case _ => None
+      }
+    }
+
+    def extractFirstParagraph(html: String) = {
+      val firstParagraph: Option[Element] = Some(doc.select("p").first())
+      firstParagraph match {
+        case Some(p) => p.text
+        case _ => ""
+      }
+    }
+
+    extractGuardianUrl map { extractedUrl =>
+
+      new Masterclass(
+        id = event.id,
+        name = event.name,
+        startDate = event.startDate,
+        url = event.url,
+        description = event.description,
+        status = event.status,
+        venue = event.venue,
+        tickets = event.tickets,
+        capacity = event.capacity,
+        guardianUrl = extractedUrl,
+        firstParagraph = extractFirstParagraph(event.description),
+        keywordIdSuffixes = Nil,
+        mainPicture = None
+      )
+    }
+  }
+
+  implicit val writesMasterclass: Writes[Masterclass] = new Writes[Masterclass] {
+    def writes(m: Masterclass) = Json.obj(
+      "id" -> m.id,
+      "name" -> m.name,
+      "startDate" -> m.startDate,
+      "url" -> m.guardianUrl,
+      "venue" -> m.venue,
+      "tickets" -> m.tickets,
+      "capacity" -> m.capacity,
+      "pictureUrl" -> m.mainPicture.map(picture => Item300.bestFor(picture.images))
+    )
+  }
+}
+
+object TravelOffer {
+
+  def fromXml(xml: Node): TravelOffer = {
+
+    def text(nodeSelector: String): String = (xml \ nodeSelector).text.trim
+    def tagText(group: String): Seq[String] = (xml \\ "tag").filter(hasGroupAttr(_, group)).map(_.text.trim)
+
+    def parseInt(s: String) = try {
+      Some(s.toInt)
```

Comment:
`scala.util.Try` does exactly this, and it useful to avoid using the low level `try` keyword;

```scala
   Try(s.toInt).toOption
```

It only catches `NonFatal`.